LearningMissionSimulation/PlaygroundEdvard.cs

```diff
@@ -316,22 +345,55 @@ public void SimulationBig1(int subjectCount, int moduleCount)
         #region Reports
         void ReportHeader(string actionName)
         {
-            Console.WriteLine($"******{actionName} is started******\n");
+            switch (this.ReportType)
+            {
+                case ReportType.Verbose:
+                case ReportType.Short:
+                    Console.WriteLine($"******{actionName} is started******\n");
+                    break;
+            }
         }
 
         void ReportFooter(string actionName)
         {
-            Console.WriteLine($"******{actionName} is finished******\n");
+            switch (this.ReportType)
+            {
+                case ReportType.Verbose:
+                case ReportType.Short:
+                    Console.WriteLine($"******{actionName} is finished******\n");
+                    break;
+            }
         }
 
         void ReportItem(string itemName, string actionName, int itemIndex)
         {
-            Console.WriteLine($" {actionName} {itemIndex}\n{itemName}");
+            switch (this.ReportType)
+            {
+                case ReportType.Verbose:
+                    Console.WriteLine($" {actionName} {itemIndex}\n{itemName}");
+                    break;
+            }
         }
 
         void ReportSummary(string summary, int count)
         {
-            Console.WriteLine($"''''''Generated {count} {summary}''''''\n");
+            switch (this.ReportType)
+            {
+                case ReportType.Verbose:
+                    Console.WriteLine($"''''''Generated {count} {summary}''''''\n");
+                    break;
+            }
+        }
+
+        void ReportError(string missingResource, string failedAction)
+        {
+            switch (this.ReportType)
```

Comment:
You should report errors in all cases but ReportType.Silent.

Reply:
Done

Reply:
it could be done in the opposite fashion:
if(this.ReportType != ReportType.Silent)
{
     // report
}

But you can keep the current implementation as more readable.